clients/js-legacy/src/actions.ts

```diff
@@ -0,0 +1,283 @@
+import {
+    createCloseContextStateInstruction,
+    createVerifyCiphertextCiphertextEqualityInstruction,
+    createVerifyCiphertextCommitmentEqualityInstruction,
+    createVerifyPubkeyValidityInstruction,
+    createVerifyZeroCiphertextInstruction,
+} from './instructions';
+import {
+    CIPHERTEXT_CIPHERTEXT_EQUALITY_CONTEXT_ACCOUNT_SIZE,
+    CIPHERTEXT_COMMITMENT_EQUALITY_CONTEXT_ACCOUNT_SIZE,
+    PUBKEY_VALIDITY_CONTEXT_ACCOUNT_SIZE,
+    ZERO_CIPHERTEXT_CONTEXT_ACCOUNT_SIZE,
+    ZK_ELGAMAL_PROOF_PROGRAM_ID
+} from './constants';
+import {
+    ElGamalCiphertext,
+    ElGamalKeypair,
+    ElGamalPubkey,
+    PedersenCommitment,
+    PedersenOpening
+} from '@solana/zk-sdk';
+
+/** Context state account information to be used as parameters to functions */
+export interface contextStateInfo {
+    /**
+     * Keypair of the context state account. If provided, use the system
+     * program to create the context state account.
+     */
+    keypair?: Signer,
+    /** Address of the context state account */
+    address: PublicKey,
+    /** Authority of the context state account */
+    authority: PublicKey,
+}
+
+/**
+ * Close a context state account
+ *
+ * @param connection                Connection to use
+ * @param payer                     Payer of the transaction fees
+ * @param contextStateAddress       Address of the context state account
+ * @param contextStateAuthority     Authority of the context state account
+ * @param destinationAccount        Destination account for the lamports
+ * @param confirmOptions            Options for confirming the transaction
+ *
+ * @return Signature of the confirmed transaction
+ */
+export async function closeContextStateProof(
+    connection: Connection,
+    payer: Signer,
+    contextStateAddress: PublicKey,
+    contextStateAuthority: Signer,
+    destinationAccount: PublicKey,
+    confirmOptions?: ConfirmOptions,
+    programId = ZK_ELGAMAL_PROOF_PROGRAM_ID,
+): Promise<TransactionSignature> {
+    const transaction = new Transaction().add(
+        createCloseContextStateInstruction(
+            contextStateAddress,
+            contextStateAuthority.publicKey,
+            destinationAccount,
+            programId,
+        )
+    );
+    return await sendAndConfirmTransaction(connection, transaction, [payer, contextStateAuthority], confirmOptions);
+}
+
+/**
+ * Verify a zero-ciphertext proof
+ *
+ * @param connection                Connection to use
+ * @param payer                     Payer of the transaction fees
+ * @param elgamalKeypair            ElGamal keypair associated with the ciphertext
+ * @param elgamalCiphertext         ElGamal encryption of zero
+ * @param contextStateInfo          Optional context state info
+ * @param confirmOptions            Options for confirming the transaction
+ *
+ * @return Signature of the confirmed transaction
+ */
+export async function verifyZeroCiphertext(
```

Comment:
Would this function be essentially useless without a context state account? If that's the case, let's make the context state account mandatory

Reply:
Yeah as of right now, this function is useless unless it is creating a context state account. Later on though, we will add support for using the record program as an alternative. In this case, it might make sense to keep the context state account optional.

But I don't quite know yet how the support for the record program would look like yet, so I did update it to make it mandatory as you suggested. Updated in [9b0b328](https://github.com/solana-program/zk-elgamal-proof/pull/1/commits/9b0b328137aae65a147f6c98a4c12173e6c2f31a).

Reply:
Yeah, I'm not sure how that would look either... we can always make a breaking change to this package without too much pain if needed